src/main/scala/PlutoUpdaterActor.scala

```diff
@@ -0,0 +1,122 @@
+import java.util.Base64
+
+import akka.actor.ActorSystem
+import akka.stream.scaladsl.{Keep, Sink, Source}
+import akka.util.ByteString
+import akka.actor.Actor
+import akka.actor.Props
+import akka.event.{DiagnosticLoggingAdapter, Logging}
+import com.gu.contentatom.thrift.{Atom, AtomData}
+import com.softwaremill.sttp._
+import com.softwaremill.sttp.akkahttp._
+import com.typesafe.config.Config
+import java.time.{Instant, LocalDateTime, ZoneOffset, ZonedDateTime}
+
+import akka.stream.ActorMaterializer
+
+import scala.concurrent.Future
+import scala.util.{Failure, Success, Try}
+
+case class DoUpdate(atom:Atom)
+case class SuccessfulSend()
+case class NoResponseBody()
+case class ErrorSend(error:String) extends Throwable {
+  override def getMessage: String = error
+}
+
+class PlutoUpdaterActor(config:Config) extends Actor{
+  implicit val sttpBackend:SttpBackend[Future,Source[ByteString, Any]] = AkkaHttpBackend.usingActorSystem(context.system)
+  import context.dispatcher
+  implicit val materializer = ActorMaterializer()
+
+  private val plutoHost=config.getString("pluto_host")
+  private val plutoPort=config.getInt("pluto_port")
+  private val plutoUser=config.getString("pluto_user")
+  private val plutoPass=config.getString("pluto_pass")
+  private val proto="http"
+
+  private implicit val logger:DiagnosticLoggingAdapter = Logging.getLogger(this)
+
+  override def receive: Receive = {
+    case DoUpdate(atom)=>
+      logger.info(s"Received update request from ${sender()}")
+      val origSender = sender() //need to make a copy of this so when it's called from another thread in map() below it is passed correctly.
+      doUpdate(atom).map({
+      case Success(futureString)=>
```

Comment:
we can use [for comprehension](https://alvinalexander.com/scala/how-use-multiple-scala-futures-in-for-comprehension-loop) for this

Reply:
i'll give it another crack. i tried for comprehensions several times in this project and never managed to make them compile

Reply:
ah, I think that this is fixed in the next PR of the sequence.  Here doUpdate has a signature of `Future[Try[Future[String]]]`, I then realised that the Future itself has the effects of a Try and simplified it down to a simple `Future[String]`